Android/app/src/main/java/app/intra/DnsVpnService.java

```diff
@@ -286,14 +286,14 @@ private void pingLocalDns() {
     }
   }
 
-  private void startDnsResolver() {
-    if (dnsResolver == null) {
+  private synchronized void startDnsResolver() {
+    if (dnsResolver == null && serverConnection != null) {
```

Comment:
Should we submit a crash report if `serverConnection == null`?

Reply:
I don't think it's necessarily an error.  For example, if `onNetworkConnected` runs during bootstrap, and bootstrap fails, then `onNetworkConnected` will call `startDnsResolver`, but `dnsResolver` and `serverConnection` will both be `null`.

It does seem like there's probably a better way, but fixing it might require rethinking the whole activation user experience.